src/preprocessed/MgvGlobal.post.sol

```diff
@@ -40,104 +28,126 @@ struct GlobalUnpacked {
 
   function unpack(GlobalPacked __packed) internal pure returns (address __monitor, bool __useOracle, bool __notify, uint __gasprice, uint __gasmax, bool __dead) { unchecked {
-    __monitor = address(uint160((GlobalPacked.unwrap(__packed) << monitor_before) >> (256-monitor_bits)));
-    __useOracle = (((GlobalPacked.unwrap(__packed) << useOracle_before) >> (256-useOracle_bits)) > 0);
-    __notify = (((GlobalPacked.unwrap(__packed) << notify_before) >> (256-notify_bits)) > 0);
-    __gasprice = (GlobalPacked.unwrap(__packed) << gasprice_before) >> (256-gasprice_bits);
-    __gasmax = (GlobalPacked.unwrap(__packed) << gasmax_before) >> (256-gasmax_bits);
-    __dead = (((GlobalPacked.unwrap(__packed) << dead_before) >> (256-dead_bits)) > 0);
+    __monitor   = address(uint160((GlobalPacked.unwrap(__packed) << monitor_before) >> (256 - monitor_bits)));
+    __useOracle = (((GlobalPacked.unwrap(__packed) << useOracle_before) >> (256 - useOracle_bits)) > 0);
+    __notify    = (((GlobalPacked.unwrap(__packed) << notify_before) >> (256 - notify_bits)) > 0);
+    __gasprice  = (GlobalPacked.unwrap(__packed) << gasprice_before) >> (256 - gasprice_bits);
+    __gasmax    = (GlobalPacked.unwrap(__packed) << gasmax_before) >> (256 - gasmax_bits);
+    __dead      = (((GlobalPacked.unwrap(__packed) << dead_before) >> (256 - dead_bits)) > 0);
   }}
 
   function monitor(GlobalPacked __packed) internal pure returns(address) { unchecked {
-    return address(uint160((GlobalPacked.unwrap(__packed) << monitor_before) >> (256-monitor_bits)));
+    return address(uint160((GlobalPacked.unwrap(__packed) << monitor_before) >> (256 - monitor_bits)));
   }}
+
   function monitor(GlobalPacked __packed,address val) internal pure returns(GlobalPacked) { unchecked {
-    return GlobalPacked.wrap((GlobalPacked.unwrap(__packed) & monitor_mask)
-                                | ((uint(uint160(val)) << (256-monitor_bits) >> monitor_before)));
+    uint __clean_struct = GlobalPacked.unwrap(__packed) & monitor_mask;
+    uint __clean_field  = (uint(uint160(val)) << (256 - monitor_bits)) >> monitor_before;
+    return GlobalPacked.wrap(__clean_struct | __clean_field);
   }}
+  
   function useOracle(GlobalPacked __packed) internal pure returns(bool) { unchecked {
-    return (((GlobalPacked.unwrap(__packed) << useOracle_before) >> (256-useOracle_bits)) > 0);
+    return (((GlobalPacked.unwrap(__packed) << useOracle_before) >> (256 - useOracle_bits)) > 0);
   }}
+
   function useOracle(GlobalPacked __packed,bool val) internal pure returns(GlobalPacked) { unchecked {
-    return GlobalPacked.wrap((GlobalPacked.unwrap(__packed) & useOracle_mask)
-                                | ((uint_of_bool(val) << (256-useOracle_bits) >> useOracle_before)));
+    uint __clean_struct = GlobalPacked.unwrap(__packed) & useOracle_mask;
+    uint __clean_field  = (uint_of_bool(val) << (256 - useOracle_bits)) >> useOracle_before;
+    return GlobalPacked.wrap(__clean_struct | __clean_field);
```

Comment:
Previously we did it all in a return, now we assign and then finally return. Have you verified gas consumption?